packages/lesswrong/components/dialogues/DialoguesList.tsx

```diff
@@ -227,6 +236,62 @@ const DialoguesList = ({ classes }: { classes: ClassesType<typeof styles> }) =>
     userDialogueChecksResult: { results: userDialogueChecks = [] },
   } = useDialogueMatchmaking({ getMatchedUsers: true, getRecommendedUsers: true, getOptedInUsers: false, getUserDialogueChecks: true });
 
+  const [upsertCheckHideInRecommendations] = useMutation(gql`
+    mutation upsertCheckHideInRecommendations($targetUserId: String!, $hideInRecommendations: Boolean!) {
+      upsertCheckHideInRecommendations(targetUserId: $targetUserId, hideInRecommendations: $hideInRecommendations) {
+          ...DialogueCheckInfo
+        }
+      }
+    ${getFragmentText('DialogueCheckInfo')}
+    ${getFragmentText('DialogueMatchPreferencesDefaultFragment')}
+    `)
+
+  const hideRecommendation = (dialogueCheckId:string|undefined, checked:boolean, targetUserId:string) => {
+    captureEvent("hide_dialogue_recommendation")
+    void upsertCheckHideInRecommendations({
+      variables: {
+        targetUserId: targetUserId, 
+        hideInRecommendations: true
+      },
+      update(cache, { data }) {
+        if (!dialogueCheckId) {
+          cache.modify({
+            fields: {
+              dialogueChecks(existingChecksRef) {
+                const newCheckRef = cache.writeFragment({
+                  data: data.upsertCheckHideInRecommendations,
+                  fragment: gql`
+                    ${getFragmentText('DialogueCheckInfo')}
+                    ${getFragmentText('DialogueMatchPreferencesDefaultFragment')}
+                  `,
+                  fragmentName: 'DialogueCheckInfo'
+                });
+                return {
+                  ...existingChecksRef,
+                  results: [...existingChecksRef.results, newCheckRef]
+                }
+              }
+            }
+          });
+        }
+      },
+      optimisticResponse: {
+        upsertCheckHideInRecommendations: {
+          _id: dialogueCheckId ?? randomId(),
+          __typename: 'DialogueCheck',
+          userId: currentUser?._id,
+          targetUserId: targetUserId,
+          checked: checked,
+          checkedAt: new Date(),
+          hideInRecommendations: true,
+          match: false,
+          matchPreference: null,
+          reciprocalMatchPreference: null
+        }
+      }
+    })
+  }
```

Comment:
Is this identical to the upsert logic used in `DialogueMatchingPage` (apart from the mutation, which I recommend that you merge with the existing one below)?  If so, given how complicated it is, it really should be factored out into a hook for re-use across multiple components